proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/client/HttpRequestHelper.java

```diff
@@ -0,0 +1,363 @@
+        /**
+         * HTTP method
+         */
+        public final @NonNull String method;
+        /**
+         * Dictionary with all HTTP headers
+         */
+        public final @NonNull Map<String, String> httpHeaders;
+        /**
+         * HTTP request body. May contain an empty array of bytes.
+         */
+        public final @NonNull byte[] body;
+
+        /**
+         * @param url full URL
+         * @param method HTTP method
+         * @param httpHeaders HTTP headers
+         * @param body HTTP request body
+         */
+        public RequestData(
+                @NonNull URL url,
+                @NonNull String method,
+                @NonNull Map<String, String> httpHeaders,
+                @Nullable byte[] body) {
+            this.url = url;
+            this.method = method;
+            this.httpHeaders = httpHeaders;
+            this.body = body;
+        }
+    }
+
+
+    /**
+     * Build data for HTTP request. The method also updates {@link #encryptor} property when the ECIES encryption
+     * is used. The encryptor can be later used for response decryption.
+     *
+     * @param baseUrl String with base URL
+     * @param helper Private cryptographic helper
+     * @return {@link RequestData} object with all information needed for request execution
+     * @throws PowerAuthErrorException if encryption or signature calculation fails.
+     * @throws MalformedURLException if cannot construct full request URL
+     */
+    @NonNull
+    public RequestData buildRequest(@NonNull String baseUrl, @NonNull IPrivateCryptoHelper helper) throws PowerAuthErrorException, MalformedURLException {
+
+        // Sanity checks
+        final boolean needsSignature = endpoint.getAuthorizationUriId() != null;
+        final boolean needsEncryption = endpoint.getEncryptorId() != ECIESEncryptorId.None;
+
+        if (needsSignature && authentication == null) {
+            throw new PowerAuthErrorException(PowerAuthErrorCodes.PA2ErrorCodeWrongParameter, "Authentication object is missing.");
+        }
+
+        // Prepare data for a new RequestData object
+
+        final URL requestUrl = new URL(baseUrl + endpoint.getRelativePath());
+        final String requestMethod = endpoint.getHttpMethod();
+        final HashMap<String, String> requestHeaders = new HashMap<String, String>();
+        final byte[] requestData;
+
+        // Encrypt the request data if the endpoint has encryptor specified
+        if (!needsEncryption) {
+            // No data encryption
+            encryptor = null;
+            requestData = serialization.serializeRequestObject(requestObject);
+        } else {
+            // Acquire the encryptor from the helper and keep it locally.
+            // We will use it later for the response decryption.
+            encryptor = helper.getEciesEncryptor(endpoint.getEncryptorId());
+            if (encryptor == null) {
+                // This typically happens when
```

Comment:
When...?

Reply:
I've changed how that encryptor is created. The helper now always returns encryptor or throws an exception in case of error.